hs/src/Reach/AST/DLBase.hs

```diff
@@ -1083,9 +1113,12 @@ instance Pretty FluidVar where
     FV_baseWaitSecs -> "baseWaitSecs"
     FV_didSend -> "didPublish"
 
+maybeTokenMap :: M.Map SLVar DLType
+maybeTokenMap = M.fromList [("None", T_Null), ("Some", T_Token)]
+
 fluidVarType :: FluidVar -> DLType
 fluidVarType = \case
-  FV_balance _ -> T_UInt
+  FV_balances i _ -> T_TokenBalances (i + 1)
```

Comment:
I thought that `FV_balances` would be of type `T_TokenBalances` (no parameter) or we could manage to avoid looking at it until later and thus never need to see the type, so this would be `impossible`